src/sonic-bgpcfgd/bgpcfgd/managers_bgp.py

```diff
@@ -324,6 +324,15 @@ def get_local_interface(self, local_addr):
                  If the interface has not been set, return None
         """
         local_addresses = self.directory.get_slot("LOCAL", "local_addresses")
+
+        # Handle local_addr string that is not represented in canonical text format, especially for IPv6 addresses
+        try:
+            ip = netaddr.IPAddress(str(local_addr))
+        except (netaddr.NotRegisteredError, netaddr.AddrFormatError, netaddr.AddrConversionError):
+            log_warn("IP Address '%s' format is wrong" % (local_addr))
+            return None
+        local_addr = str(ip)
+
```

Comment:
@chenkelly , can you also add unit test for this code?

Reply:
Ok, I'll check and see how to add it.